rust/datafusion/src/physical_plan/union.rs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Some of these functions reference the Postgres documentation
+// or implementation to ensure compatibility and are subject to
+// the Postgres license.
+
+//! Unicode expressions
+
+use std::{any::Any, sync::Arc};
+
+use arrow::datatypes::SchemaRef;
+
+use super::{ExecutionPlan, Partitioning, SendableRecordBatchStream};
+use crate::error::Result;
+use async_trait::async_trait;
+
+/// UNION ALL execution plan
+#[derive(Debug)]
+pub struct UnionExec {
+    /// Input execution plan
+    inputs: Vec<Arc<dyn ExecutionPlan>>,
+}
+
+impl UnionExec {
+    /// Create a new MergeExec
+    pub fn new(inputs: Vec<Arc<dyn ExecutionPlan>>) -> Self {
+        UnionExec { inputs }
+    }
+}
+
+#[async_trait]
+impl ExecutionPlan for UnionExec {
+    /// Return a reference to Any that can be used for downcasting
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn schema(&self) -> SchemaRef {
+        self.inputs[0].schema()
+    }
+
+    fn children(&self) -> Vec<Arc<dyn ExecutionPlan>> {
+        self.inputs.clone()
+    }
+
+    /// Get the output partitioning of this plan
+    fn output_partitioning(&self) -> Partitioning {
+        Partitioning::UnknownPartitioning(self.inputs.len())
+    }
+
+    fn with_new_children(
+        &self,
+        children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        Ok(Arc::new(UnionExec::new(children)))
+    }
+
+    async fn execute(&self, partition: usize) -> Result<SendableRecordBatchStream> {
+        self.inputs[partition].execute(0).await
```

Comment:
This seems to imply that the input plans can only have a single partition?

Reply:
I believe that was "solved" by adding a `mergeexec` but I will probably change this part and add some tests.

Reply:
I removed the `MergeExec` and changed this to what I believe Spark also does, which is return the sum of the input partitions and added a test for this.

For now, this removes the information about the input partitioning and return `UnknownPartitioning`, which is not ideal when the partitioning info could be used somewhere else (for example a hash partitioning).
I added a TODO comment there. I think it would be OK to keep it for now like this and maybe create a JIRA issue for?

Reply:
Thanks, this is looking good.